apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/TableStatsCollectorUtil.java

```diff
@@ -114,16 +119,42 @@ protected static IcebergTableStats populateStatsForSnapshots(
             .min(Long::compareTo)
             .orElse(null);
 
+    Map<String, Long> snapshotCountByDay =
+        getSnapShotDistributionPerDay(table, spark, MetadataTableType.SNAPSHOTS);
+
     return stats
         .toBuilder()
         .currentSnapshotId(currentSnapshotId)
         .currentSnapshotTimestamp(currentSnapshotTimestamp)
         .oldestSnapshotTimestamp(oldestSnapshotTimestamp)
         .numCurrentSnapshotReferencedDataFiles(countOfDataFiles)
         .totalCurrentSnapshotReferencedDataFilesSizeInBytes(sumOfFileSizeBytes)
+        .snapshotCountByDay(snapshotCountByDay)
         .build();
   }
 
+  /** Get snapshot distribution for a given table by date. */
+  private static Map<String, Long> getSnapShotDistributionPerDay(
+      Table table, SparkSession spark, MetadataTableType metadataTableType) {
+    Dataset<Row> snapShotDistribution =
+        SparkTableUtil.loadMetadataTable(spark, table, metadataTableType)
```

Comment:
Will this get all snapshots committed from beginning of table?
If yes, should it have a filter criteria as well to get snapshots count only in last `X` days?